integration/testrunner/db.go

```diff
@@ -16,45 +16,62 @@ package testrunner
 
 import (
 	"context"
+	"errors"
 	"fmt"
+	"net/mail"
 	"testing"
 
 	"cloud.google.com/go/bigquery"
+	"cloud.google.com/go/logging/apiv2/loggingpb"
 	"github.com/sethvargo/go-retry"
+	"google.golang.org/api/iterator"
+	"google.golang.org/genproto/googleapis/cloud/audit"
+	"google.golang.org/protobuf/encoding/protojson"
 )
 
-// queryIfAuditLogExists queries the DB and checks if audit log contained in the query exists or not.
-func queryIfAuditLogExists(ctx context.Context, tb testing.TB, query *bigquery.Query, expectedNum int64) (bool, error) {
+// queryAuditLogs queries the DB and checks if audit log contained in the query exists or not and return the results.
+// For retryable cases, return error. For non-retryable cases, fail now.
+func queryAuditLogs(ctx context.Context, tb testing.TB, query *bigquery.Query) ([]*loggingpb.LogEntry, error) {
```

Comment:
A second thought on this: I wonder if we can skip the error return to make the code cleaner.

If we hit an error that's retryable, we just `t.Log` it and return nil. For non-retryable logic, we will do the `t.Fatal` as you're already doing.

Reply:
It makes sense only when this method is only used by `validateAuditLogsWithRetries`. Though currently only`validateAuditLogsWithRetries` calls `queryAuditLogs`, I'm not sure whether in the future we will have other callers. I don't have strong preference now. I will update it and add comments to clarify it. But if in the future, we have other callers of this method, we need to keep it in mind.

Reply:
done